src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs

```diff
@@ -1,14 +1,82 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Reflection;
 using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
 
 namespace System.Runtime.InteropServices.JavaScript
 {
     // this maps to src\mono\wasm\runtime\corebindings.ts
     // the methods are protected from trimming by DynamicDependency on JSFunctionBinding.BindJSFunction
     internal static unsafe partial class JavaScriptExports
     {
+        [MethodImpl(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
+        // the marshaled signature is:
+        // Task<int>? CallEntrypoint(MonoMethod* entrypointPtr, string[] args)
+        public static void CallEntrypoint(JSMarshalerArgument* arguments_buffer)
+        {
+            ref JSMarshalerArgument arg_exc = ref arguments_buffer[0]; // initialized by caller in alloc_stack_frame()
+            ref JSMarshalerArgument arg_result = ref arguments_buffer[1]; // initialized by caller in alloc_stack_frame()
+            ref JSMarshalerArgument arg_1 = ref arguments_buffer[2]; // initialized and set by caller
+            ref JSMarshalerArgument arg_2 = ref arguments_buffer[3]; // initialized and set by caller
+            try
+            {
+                arg_1.ToManaged(out IntPtr entrypointPtr);
+                arg_2.ToManaged(out string?[]? args);
+                RuntimeMethodHandle methodHandle = JSHostImplementation.GetMethodHandleFromIntPtr(entrypointPtr);
+                MethodInfo? method = MethodBase.GetMethodFromHandle(methodHandle) as MethodInfo;
```

Comment:
this is only valid if the type isn't generic, which i think is prohibited for Main, so that's fine. might be worth adding a comment to specify that it's safe to use this overload here for that reason